glue/core/data.py

```diff
@@ -1661,8 +1661,53 @@ def compute_statistic(self, statistic, cid, subset_state=None, axis=None,
                 data = subset_state.to_array(self, cid)
             else:
                 mask = subset_state.to_mask(self, view)
+
                 if np.any(unbroadcast(mask)):
+
+                    # Find minimal subarray containing the masked area.
+                    subarray_slices = []
+                    for idim in range(mask.ndim):
+                        collapse_axes = tuple(index for index in range(mask.ndim) if index != idim)
+                        valid = mask.any(axis=collapse_axes)
```

Comment:
This step can be arbitrarily expensive, no?  It will loop over all of the mask data?  Or is mask already sliced down at a previous step?

Reply:
We already access np.any on the mask above so this won't change much, and we typically operate in chunks for big arrays.